crates/nu-protocol/src/eval_const.rs

```diff
@@ -26,36 +26,37 @@ pub fn create_nu_constant(engine_state: &EngineState, span: Span) -> Result<Valu
 
     let mut record = Record::new();
 
+    let config_path = match nu_path::config_dir() {
+        Some(mut path) => {
+            path.push("nushell");
+            Ok(path)
+        }
+        None => Err(Value::error(
+            ShellError::ConfigDirNotFound { span: Some(span) },
+            span,
+        )),
+    };
+
     record.push(
         "default-config-dir",
-        if let Some(mut path) = nu_path::config_dir() {
-            path.push("nushell");
-            Value::string(path.to_string_lossy(), span)
-        } else {
-            Value::error(
-                ShellError::IOError {
-                    msg: "Could not get config directory".into(),
-                },
-                span,
-            )
-        },
+        config_path.as_ref().map_or_else(
```

Comment:
i think i understand the logic behind that, but i'm wondering if these `map_or_else` are required :thinking: 

why not replace the `Ok(path)` in the `let config_path` above with `Value::string(path.to_string_lossy(), span)` directly?
i feel that would allow to directly use `config_path` or maybe `config_path.clone()` in these `record.push`, rigth? :innocent:

Reply:
@amtoine I might be misinterpreting you, but if the path is turned to a `Value` right there, wouldn't it have to be turned back into a `PathBuf` later so that it can be modified?

Reply:
aah yeah, i see, you need to push things like `config.nu` or `env.nu` to the `config_path` each time :thinking: 
so you'd have to unpack the `Value`, which would be a pain :eyes: 

that's fine then :smile: